It won't compile in Ubuntu 20.04

Neither version 20.0.0 or the most recent commit compile in Ubuntu 20.04 (or derivatives).

When executing
```bash
./configure --prefix=/opt/pbs
```
you get:
```
configure: error: Python must be version 3.5
else
  3.6 or 3.7
```

Python 3 in Ubuntu 20.04 is version 3.8.2. If the configure file is edited to accept Python 3.8 them the `make` command fails, with:
```
/usr/bin/ld: openpbs/src/lib/Libpython/pbs_ifl_wrap.c:2735: undefined reference to `PyUnicode_DecodeUTF8'
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:790: pbs_server.bin] Error 1
make[2]: Leaving directory 'openpbs/github/openpbs/src/server'
make[1]: *** [Makefile:511: all-recursive] Error 1
make[1]: Leaving directory 'openpbs/src'
make: *** [Makefile:547: all-recursive] Error 1
```

Given that 20.04 is the new LTS version of Ubuntu, it would be great that openpbs could be use in it.

Thank you.

--- a/configure.ac
+++ b/configure.ac
@@ -187,7 +187,6 @@ AC_CHECK_HEADERS([ \
 	sys/statfs.h \
 	sys/stat.h \
 	sys/statvfs.h \
-	sys/sysctl.h \
 	sys/sysinfo.h \
 	sys/sysmacros.h \
 	sys/time.h \

--- a/m4/with_python.m4
+++ b/m4/with_python.m4
@@ -51,15 +51,19 @@ AC_DEFUN([PBS_AC_WITH_PYTHON],
     [PYTHON_CONFIG="python3-config"]
   )
   AM_PATH_PYTHON([3.5])
-  AS_IF([test "$PYTHON_VERSION" != "3.5" -a "$PYTHON_VERSION" != "3.6" -a "$PYTHON_VERSION" != "3.7"],
-    AC_MSG_ERROR([Python must be version 3.5, 3.6 or 3.7]))
-  [PYTHON_INCLUDES=`$PYTHON_CONFIG --includes`]
+  AS_IF([test "$PYTHON_VERSION" != "3.5" \
+          -a "$PYTHON_VERSION" != "3.6" \
+          -a "$PYTHON_VERSION" != "3.7" \
+          -a "$PYTHON_VERSION" != "3.8" ],
+    AC_MSG_ERROR([Python must be version 3.5, 3.6, 3.7 or 3.8]))
+  AS_IF([test "$PYTHON_VERSION" == "3.8"], [_extra_arg="--embed"], [_extra_arg=""])
+  [PYTHON_INCLUDES=`$PYTHON_CONFIG --includes ${_extra_arg}`]
   AC_SUBST(PYTHON_INCLUDES)
-  [PYTHON_CFLAGS=`$PYTHON_CONFIG --cflags`]
+  [PYTHON_CFLAGS=`$PYTHON_CONFIG --cflags ${_extra_arg}`]
   AC_SUBST(PYTHON_CFLAGS)
-  [PYTHON_LDFLAGS=`$PYTHON_CONFIG --ldflags`]
+  [PYTHON_LDFLAGS=`$PYTHON_CONFIG --ldflags ${_extra_arg}`]
   AC_SUBST(PYTHON_LDFLAGS)
-  [PYTHON_LIBS=`$PYTHON_CONFIG --libs`]
+  [PYTHON_LIBS=`$PYTHON_CONFIG --libs ${_extra_arg}`]
   AC_SUBST(PYTHON_LIBS)
   AC_DEFINE([PYTHON], [], [Defined when Python is available])
   AC_DEFINE_UNQUOTED([PYTHON_BIN_PATH], ["$PYTHON"], [Python executable path])

--- a/m4/with_swig.m4
+++ b/m4/with_swig.m4
@@ -56,5 +56,9 @@ AC_DEFUN([PBS_AC_WITH_SWIG],
     AC_DEFINE([SWIG], [], [Defined when swig is available]),
     AC_MSG_RESULT([not found])
     AC_MSG_WARN([swig command not found.]))
+  AS_IF([test "x`ls -d ${swig_dir}/share/swig/* 2>/dev/null`" == "x" ],
+          [swig_py_inc="-I`ls -d ${swig_dir}/share/swig*` -I`ls -d ${swig_dir}/share/swig*/python`"],
+          [swig_py_inc="-I`ls -d ${swig_dir}/share/swig/*` -I`ls -d ${swig_dir}/share/swig/*/python`"])
   AC_SUBST([swig_dir])
+  AC_SUBST([swig_py_inc])
 ])

--- a/src/lib/Libpython/Makefile.am
+++ b/src/lib/Libpython/Makefile.am
@@ -88,7 +88,6 @@ pbs_ifl_wrap.c: $(top_srcdir)/src/lib/Libifl/pbs_ifl_wrap.c
 	echo '#include "pbs_ifl.h"' >> pbs_ifl.i ; \
 	echo '%}' >> pbs_ifl.i ; \
 	echo '%include "pbs_ifl.h"' >> pbs_ifl.i ; \
-	@swig_dir@/bin/swig -I`ls -d @swig_dir@/share/swig/*` \
-		-I`ls -d @swig_dir@/share/swig/*/python` \
+	@swig_dir@/bin/swig @swig_py_inc@ \
 		-I$(top_srcdir)/src/include \
 		-python pbs_ifl.i

--- a/src/lib/Libpython/common_python_utils.c
+++ b/src/lib/Libpython/common_python_utils.c
@@ -72,7 +72,7 @@ void
 pbs_python_write_object_to_log(PyObject *obj, char *pre, int severity)
 {
 	PyObject *py_tmp_str = NULL;
-	char *obj_str = NULL;
+	const char *obj_str = NULL;
 
 	if (!(py_tmp_str = PyObject_Str(obj))) { goto ERROR_EXIT; }
 	if (!(obj_str = PyUnicode_AsUTF8(py_tmp_str))) { goto ERROR_EXIT; }
@@ -432,7 +432,7 @@ pbs_python_object_get_attr_integral_value(PyObject *obj, const char *key)
 char *
 pbs_python_object_str(PyObject *obj)
 {
-	char *str = NULL;
+	const char *str = NULL;
 	PyObject *py_str;
 	static char *ret_str = NULL;
 	char  	*tmp_str = NULL;

--- a/src/lib/Libpython/pbs_python_external.c
+++ b/src/lib/Libpython/pbs_python_external.c
@@ -698,7 +698,7 @@ pbs_python_run_code_in_namespace(struct python_interpreter_data *interp_data,
 	PyObject *ptraceback;
 	PyObject *pobjStr;
 	PyObject *retval;
-	char      *pStr;
+	const char      *pStr;
 	int rc=0;
 	pid_t orig_pid;
 

--- a/src/lib/Libpython/pbs_python_svr_size_type.c
+++ b/src/lib/Libpython/pbs_python_svr_size_type.c
@@ -124,7 +124,7 @@ extern PyObject * PPSVR_Size_FromSizeValue(struct size_value);
 static int
 _pps_check_for_negative_number(PyObject *il) {
 	PyObject *str_value = NULL;
-	char *c_value;
+	const char *c_value;
 	int rc = 0;
 
 	if (!(str_value = PyObject_Str(il))) {
@@ -247,7 +247,7 @@ _pps_size_from_string(PyObject *self, PyObject *from)
 
 	PPSVR_Size_Object *working_copy = (PPSVR_Size_Object *) self;
 	if (PyUnicode_Check(from)) {
-		if ((to_size(PyUnicode_AsUTF8(from), &working_copy->sz_value) != 0)) {
+		if ((to_size((char *)PyUnicode_AsUTF8(from), &working_copy->sz_value) != 0)) {
 			snprintf(log_buffer, LOG_BUF_SIZE-1, "%s: bad value for _size",
 				pbs_python_object_str(from));
 			PyErr_SetString(PyExc_TypeError, log_buffer);

--- a/src/resmom/catch_child.c
+++ b/src/resmom/catch_child.c
@@ -467,7 +467,7 @@ json_dumps(PyObject *py_val, char *msg, size_t msg_len)
 	PyObject	*py_func_dumps = NULL;
 	PyObject	*py_value = NULL;
 	PyObject	*py_result = NULL;
-	char		*tmp_str = NULL;
+	const char	*tmp_str = NULL;
 	char		*ret_string = NULL;
 	int		slen;
 

--- a/src/scheduler/fifo.c
+++ b/src/scheduler/fifo.c
@@ -149,7 +149,7 @@ schedinit(int nthreads)
 	struct tm *tmptr;
 
 #ifdef PYTHON
-	char *errstr;
+	const char *errstr;
 	PyObject *module;
 	PyObject *obj;
 	PyObject *dict;

--- a/src/scheduler/job_info.c
+++ b/src/scheduler/job_info.c
@@ -4296,7 +4296,7 @@ formula_evaluate(char *formula, resource_resv *resresv, resource_req *resreq)
 	int globals_size = 1024;  /* initial size... will grow if needed */
 	resource_req *req;
 	sch_resource_t ans = 0;
-	char *str;
+	const char *str;
 	int i;
 	char *formula_buf;
 	int formula_buf_len;

--- a/src/server/vnparse.c
+++ b/src/server/vnparse.c
@@ -258,9 +258,7 @@ vn_parse_stream(FILE *fp, callfunc_t callback)
 		for (vnp = vnid; *vnp && legal_vnode_char(*vnp, 1); vnp++)
 			;
 		if (*vnp) {
-			sprintf(log_buffer,
-				"invalid character in vnode name \"%s\"", vnid);
-			log_err(PBSE_SYSTEM, __func__, log_buffer);
+			log_errf(PBSE_SYSTEM, __func__, "invalid character in vnode name \"%s\"", vnid);
 			vnl_free(vnlp);
 			return NULL;
 		}
@@ -270,9 +268,7 @@ vn_parse_stream(FILE *fp, callfunc_t callback)
 		 * is defined as string of length 64.
 		 */
 		if (strlen(vnid) > PBS_MAXHOSTNAME) {
-			sprintf(log_buffer,
-				"Node name \"%s\" is too big", vnid);
-			log_err(PBSE_SYSTEM, __func__, log_buffer);
+			log_errf(PBSE_SYSTEM, __func__, "Node name \"%s\" is too big", vnid);
 			return NULL;
 		}
 		/* <ATTRNAME> <ATTRDELIM> */

